src/logfile_values_experimental.py

```diff
@@ -0,0 +1,224 @@
+        if not i % 2:
+            return (sorted_list[(i // 2) - 1] + sorted_list[i // 2]) / 2
+        return sorted_list[i // 2]
+
+    def get_sum(self):
+        return sum(self.values)
+
+    def get_count(self, v=0):
+        return sum(1 for x in self.values if x > v)
+
+    def get_mean(self):
+        return sum(self.values) / len(self.values)
+
+    def get_min(self):
+        return min(self.values)
+
+    def get_max(self):
+        return max(self.values)
+
+    def get_last_value(self):
+        return self.last_value
+
+    def get_metric_value(self):
+        if 'count_' in self.function:
+            return int(self.get_count(int(self.function.split('_')[1])))
+        return int(getattr(self, 'get_' + self.function)())
+
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument('--prefix', default='logfile_values')
+    parser.add_argument('--file', default='/var/log/messages')
+    parser.add_argument('--metric', type=Metric, nargs='+')
+    parser.add_argument('--time_format', default='%Y-%m-%dT%H:%M:%S')
+    parser.add_argument('--arch', action='store_true')
+    parser.add_argument('--debug', '-d', action='store_true')
+    parser.add_argument('--timezone', default='Europe/Berlin')
+    return parser.parse_args()
+
+
+def yield_metric_value(line, metric, time_format):
+    fields = line.split()
+    fields[0] = convert_to_timestamp(fields[0], time_format)
+    if fields[0] > (metric.now - metric.get_timeshift()):
+        return fields[metric.column]
+    else:
+        return False
+
+
+def read_logfile_reverse(filename, time_format, metric, buf_size=8192):
+    """
+    A generator that returns the lines of a file in reverse order.
+    Stops to read when has met timestamp bigger then desired period 
+    
+    returns False if file was not read completely
+    """
+    with open(filename) as fh:
+        segment = None
+        offset = 0
+        fh.seek(0, os.SEEK_END)
+        file_size = remaining_size = fh.tell()
+        while remaining_size > 0:
+            offset = min(file_size, offset + buf_size)
+            fh.seek(file_size - offset)
+            buffer = fh.read(min(remaining_size, buf_size))
+            remaining_size -= buf_size
+            lines = buffer.splitlines()
+            # the first line of the buffer is probably not a complete line so
+            # we'll save it and append it to the last line of the next buffer
+            # we read
+            if segment is not None:
+                # if the previous chunk starts right from the beginning of line
+                # do not concact the segment to the last line of new chunk
+                # instead, yield the segment first
+                if buffer[-1] is not '\n':
+                    lines[-1] += segment
+                else:
+                    yield yield_metric_value(segment, metric, time_format)
+            segment = lines[0]
+            for index in range(len(lines) - 1, 0, -1):
```

Comment:
I would write this as `reversed(range(len(lines)))`.

Reply:
Need to exclude zero